src/main/kotlin/de/docs_as_co/intellij/plugin/drawio/editor/DiagramsEditor.kt

```diff
@@ -36,21 +23,29 @@ class DiagramsEditor(private val project: Project, private val file: VirtualFile
     private val view = DrawioWebView(lifetime)
 
     init {
+
         view.initializedPromise.then {
             view.loadXmlLike(file.inputStream.reader().readText())
         }
 
         view.xmlContent.advise(lifetime) { xml ->
             if (xml !== null) {
-                ApplicationManager.getApplication().invokeLater {
-                    ApplicationManager.getApplication().runWriteAction {
-                        file.getOutputStream(this).apply {
-                            writer().apply {
-                                write(xml)
+                val isSVGFile = file.name.endsWith(".svg")
+                val isSVGXML = xml.startsWith("<svg")
+                if ( isSVGFile && (!isSVGXML) ) {
+                    //ignore the xml payload and ask for an exported svg
+                    view.exportSvg()
```

Comment:
use `view.exportSvg.then { svg -> ... }` to save the svg. The xmlContent should always be plain drawio xml!

Reply:
so, ich glaube, das ist jetzt auch geschafft...